### PR TITLE
Tightened OspreySharp Gauss-Jordan robustness (abs pivot + tolerance)

### PR DESCRIPTION
**Pull request by brendanx67**
## Summary

Addresses three Copilot threads deferred from #4155 (parity-affecting; requires a coordinated Rust PR). Paired with **maccoss/osprey#15**, which both must be CI-green before merging together.

* `Echelon` pivot selection now uses `Math.Abs` so a large-magnitude negative is preferred over a small positive — signed-max could skip the correct row entirely. Zero-pivot guard is `|pivot| < 1E-12` instead of exact equality. ([discussion_r3111986866](https://github.com/ProteoWizard/pwiz/pull/4155#discussion_r3111986866))
* `LeftSolved` uses a 1E-8 tolerance on both diagonal (`|x - 1| < TOL || |x| < TOL`) and off-diagonal (`|x| < TOL`) checks, instead of exact `==` on diagonals and a signed `x > 1E-8` off-diagonal check that accepted arbitrarily negative residuals. ([discussion_r3111986886](https://github.com/ProteoWizard/pwiz/pull/4155#discussion_r3111986886))
* Added two regression tests in `MLTest.cs` under a new "Gauss Solver Tests" region using the same numeric inputs as the Rust `gauss_negative_pivot` and `gauss_near_singular_returns_none` tests. The negative-pivot test fails against the old echelon (verified by temporarily reverting the fix locally). ([discussion_r3111987020](https://github.com/ProteoWizard/pwiz/pull/4155#discussion_r3111987020))
* Promoted `SolveInner` from `private` to `internal` so the near-singular test can pass `eps = 0` directly, symmetric with Rust's `pub fn solve_inner`.

### Parity-affecting

`GaussSolver.Solve` is on the LDA training path (`Sw * x = Sb`), so changing pivoting or tolerance would change LDA output on inputs where the decision diverges. The Rust and C# sides land together to keep the tools in lockstep.

### Verification

* `Build-OspreySharp.ps1 -RunInspection -RunTests` — 216/216 pass, ReSharper green.
* Cross-tool parity with the paired Rust PR: Stellar (317 842 entries) and Astral (1 051 741 entries) at 1e-6, **21/21** PIN features on both datasets.
* **Fix-impact measurement**: with the C# fix stashed, re-ran the same parity tests against the already-patched Rust binary. Still 21/21 on both datasets. So on these production inputs the fix has **no observable effect** — this is defensive hygiene for corner cases (large-magnitude negative pivots, near-singular matrices) outside the current test data. The Stellar parity breakage observed during Session 20 of the #4155 work (which led to deferring these threads) was therefore misattributed to this change.

## Test plan

- [x] TestGaussNegativePivot — regression test for abs-pivot selection
- [x] TestGaussNearSingularReturnsNull — regression test for rank-deficient input
- [x] MLTest.cs full suite (216 tests)
- [x] Build-OspreySharp.ps1 -RunInspection (ReSharper green)
- [x] Test-Features.ps1 -Dataset Stellar (21/21 at 1e-6)
- [x] Test-Features.ps1 -Dataset Astral (21/21 at 1e-6)
- [x] Fix-impact measurement: stashed C# fix, re-ran Stellar + Astral (21/21 at 1e-6)

Co-Authored-By: Claude <noreply@anthropic.com>

### CODE REVIEW

File: pwiz_tools/OspreySharp/OspreySharp.ML/GaussSolver.cs [hunk 1]
```diff
@@ -112,10 +115,10 @@ private bool LeftSolved()
                     double x = _left[i, j];
                     if (i == j)
                     {
-                        if (x != 1.0 && x != 0.0)
+                        if (Math.Abs(x - 1.0) > TOL && Math.Abs(x) > TOL)
                             return false;
```
[the remark this answers is not in the document]
Comment (brendanx67) [reply pwiz_tools/OspreySharp/OspreySharp.ML/GaussSolver.cs:119]: This is not a single-line if, which would be "if (Math.Abs(x - 1.0) > TOL && Math.Abs(x) > TOL) return false;" Project convention does not require all ifs have brackets as is being suggested.
